jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpConnection.java

```diff
@@ -398,6 +398,15 @@ public void onFillable()
                     if (LOG.isDebugEnabled())
                         LOG.debug("HANDLE {} {}", request, this);
 
+                    // If the buffer is empty and no body is expected, then release the buffer
+                    if (isRequestBufferEmpty() && !_parser.hasContent())
+                    {
+                        // try parsing now to the end of the message
+                        parseRequestBuffer();
```

Comment:
We need to allow the parser to general endContent and endMessage events.  Normally done in the parseAndFill after handling, but we can do here to ensure message is really complete before releasing buffer (i.e. no trailers).